drivers/usb/uhc/uhc_mcux_common.c

```diff
@@ -243,10 +284,21 @@ static usb_host_pipe_handle uhc_mcux_check_hal_ep(const struct device *dev,
 		}
 	}
 
-	/* TODO: need to check endpoint type too */
-	if (mcux_ep != NULL &&
+	if (xfer->type == USB_EP_TYPE_INTERRUPT) {
+		if (xfer->udev->speed == USB_SPEED_SPEED_HS) {
+			actual_interval = (uint16_t)(1UL << (xfer->interval - 1U));
+		} else {
+			actual_interval = uhc_get_power_of_2_floor(xfer->interval);
+		}
+	} else if (xfer->type == USB_EP_TYPE_ISO) {
+		actual_interval = (uint16_t)(1UL << (xfer->interval - 1U));
+	} else {
+		actual_interval = xfer->interval;
+	}
```

Comment:
Suggest to add one struct member in `struct uhc_mcux_data` as follow. `mcux_ep->interval` is different with the interval value of endpoint descriptor.
```
	usb_host_pipe_t *mcux_eps[USB_HOST_CONFIG_MAX_PIPES];
    uint16_t mcux_eps_interval[USB_HOST_CONFIG_MAX_PIPES];
```